### ISSUE TITLE
The extension doesn't respect configuration change token

### ISSUE THREAD

**Issue by MichaCo** [serilog/serilog-extensions-logging-file#2]
Hi,
It seems that the change token handling isn't implemented in this extension. At least I couldn't find it in code and I found it while playing with it and changing configuration but nothing happend ;)

Is this something you are going to add?
thanks
M

**Comment by nblumhardt** [comment]
Hi Micha- thanks for the note. It seems like this would be a worthwhile addition, but there's no work in progress on it currently.

**Comment by szahn** [comment]
+1

**Comment by SixtoSaez** [comment]
Not a lot of context to this old open issue. Willing to work on it but need a few more specifics.

**Comment by nblumhardt** [comment]
Hi @SixtoSaez - thanks for jumping in. https://docs.microsoft.com/en-us/aspnet/core/fundamentals/change-tokens?view=aspnetcore-3.1 is what the OP is referring to. Shimming it in here will require some kind of sink wrapper that's capable of switching out file sinks under the hood in response to configuration changes - a proof of concept might use _Serilog.Sinks.Map_ with an open sink limit of 1 to achieve this. Hope this helps!

**Comment by SixtoSaez** [comment]
Yes it does! I'll get up to date on the change tokens and the code you suggested.
